cmd/app/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"flag"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	configFileKey     = "configFile"
+	defaultConfigFile = "config/config.yml"
+	configFileUsage   = "config file path"
+)
+
+func main() {
+	var configFile string
+	flag.StringVar(&configFile, configFileKey, defaultConfigFile, configFileUsage)
+	flag.Parse()
+
+	log.Println("configFile: ", configFile)
+
+	execute(configFile)
```

Comment:
How do you feel about returning an error and then exiting here?
I think that the function execute should not stop the program. The main function should be responsible for that. If `execute` encounters an error, we return it, and from the main function we should exit. The idea is that it is clearer from readability perspective, when you look at the main function, to know when and how it exits. See how k8s does it: https://github.com/kubernetes/kubernetes/blob/6e3d62ca1c9e1132be5cd0be6746512fdd411b60/cmd/kube-apiserver/apiserver.go#L32

Reply:
Really good suggestion, It does make it clearer, I have done it this way before but missed the return in here.